In [7]:
import json
import pandas as pd

In [8]:
indian_movies = pd.read_csv("dataset/indian_movies_dummy.csv")
indian_movies

imdb_title_id                           title  \
0        tt0116391                            Gang   
1        tt0116748  Karobaar: The Business of Love   
2        tt0118578                          Albela   
3        tt0120117                         Shikari   
4        tt0151150                            Josh   
...            ...                             ...   
5171     tt9900782                          Kaithi   
5172     tt9903716                          Jessie   
5173     tt9905412                           Ottam   
5174     tt9905462                       Pengalila   
5175     tt9906644                       Manoharam   

                      original_title  year date_published  \
0                               Gang  2000     2000-04-21   
1     Karobaar: The Business of Love  2000     2000-09-15   
2                             Albela  2001     2001-04-20   
3                            Shikari  2000     2000-10-06   
4                               Josh  2000     2000-06-09   
...                              ...   ...            ...   
5171                          Kaithi  2019     2019-10-25   
5172                          Jessie  2019     2019-03-15   
5173                           Ottam  2019     2019-03-08   
5174                       Pengalila  2019     2019-03-08   
5175                       Manoharam  2019     2019-09-27   

                       genre  duration country   language          director  \
0       Action, Crime, Drama       152   India      Hindi       Mazhar Khan   
1             Drama, Romance       180   India      Hindi     Rakesh Roshan   
2     Comedy, Drama, Romance       145   India      Hindi     Deepak Sareen   
3     Crime, Drama, Thriller       160   India      Hindi        N. Chandra   
4     Action, Drama, Musical       162   India      Hindi      Mansoor Khan   
...                      ...       ...     ...        ...               ...   
5171        Action, Thriller       145   India      Tamil  Lokesh Kanagaraj   
5172        Horror, Thriller       106   India     Telugu   Aswani Kumar V.   
5173                   Drama       120   India  Malayalam               Zam   
5174                   Drama       111   India  Malayalam     T.V. Chandran   
5175           Comedy, Drama       122   India  Malayalam       Anvar Sadik   

      ... History Biography Family Mystery  Horror  Music War Sport Sci-Fi  \
0     ...       0         0      0       0       0      0   0     0      0   
1     ...       0         0      0       0       0      0   0     0      0   
2     ...       0         0      0       0       0      0   0     0      0   
3     ...       0         0      0       0       0      0   0     0      0   
4     ...       0         0      0       0       0      0   0     0      0   
...   ...     ...       ...    ...     ...     ...    ...  ..   ...    ...   
5171  ...       0         0      0       0       0      0   0     0      0   
5172  ...       0         0      0       0       1      0   0     0      0   
5173  ...       0         0      0       0       0      0   0     0      0   
5174  ...       0         0      0       0       0      0   0     0      0   
5175  ...       0         0      0       0       0      0   0     0      0   

      Animation  
0             0  
1             0  
2             0  
3             0  
4             0  
...         ...  
5171          0  
5172          0  
5173          0  
5174          0  
5175          0  

[5176 rows x 42 columns]

Creating tfidf sparse matrix for the description

In [9]:
from sklearn.feature_extraction.text import TfidfVectorizer

tfidf = TfidfVectorizer()
description =  tfidf.fit_transform(indian_movies["description"])
description

<5176x12274 sparse matrix of type '<class 'numpy.float64'>'
	with 113162 stored elements in Compressed Sparse Row format>

Converting the matirx to array

In [10]:
description =description.toarray()
description.shape

(5176, 12274)

AS there are 5176 rows as per the movies present and for each movie there are 12k columns for denoting the description

Storing these description array to dict where keys are title ids

In [11]:
description_movie = {}
for i in range(description.shape[0]):
    title = indian_movies["imdb_title_id"].iloc[i]
    descriprion_list = description[i].tolist()
    description_movie[title] = descriprion_list
len(description_movie.keys())

5176

Keys length and number of movies are same so dict is created properly

Now dumping this file to json file for recommendation purpose

In [12]:
with open("description_movie.json", "w") as dm:
    json.dump(description_movie, dm)

##### Recommendation functions file: [Recommendaion functions file](recommendation_functions.py)
##### Recommendation execution file: [Recommendation execution ipynb file](recommend.ipynb) & [Recommendation execution html file](recommend.html)